### DISCUSSION TITLE
[Working Contract]: Holiday workflow gets wrong status

### DISCUSSION THREAD

**Discussion by idemus** [kimai/Paid-plugins#211] [General]
Hi Kevin,
thank you for implementing the holiday workflow. It works fine except of the case that the requested holiday will be declined.
Then the state gets back to 'new' instead it is switched to 'not approved'.
Could you check that, please?

Installed: Kimai version 2.26, plugin version: 1.17

Best regards
Klaus

**Comment by kevinpapst** [comment]
They get "rejected" and that state is reflected under "new".
Obviously you would have to discuss a rejected item internally and then the employee or supervisor either deletes it or it gets re-requested.

So for now it works as expected and there is nothing to check from my endpoint.

Leaving this open to collect further feedback and ideas what should happen instead.

**Comment by idemus** [comment]
Hi Kevin,

but that behaves like an 'endless' loop when I do not contact the employee before and he will push the holiday again from 'new' to 'requested'.
From  my point of view it would be very good when:
- the employee could see the denied holiday requests in his overview (so he won't create it again)
- there shouldn't be a media break in the workflow and extra communication when denying a request
- a rejected request is something else than e.g. a wrongly created holiday request (which should be permanently deleted from system as if it had never been created)

**Reply by kevinpapst** [reply]
This is likely only my limited view without the experience of having multiple employees:

The employee receives an email that the request was rejected.
If he then requests it again without talking first, isn't that in first place a communication issue instead of a software question?

Having a new table with rejected requests isn't a problem, but then what? 
Do they stay there forever? 
Can they be resubmitted? If not: why not? Because there are valid reasons like "ups, accidentally clicked Reject" or "circumstances changed, you can take a vacation".
Can they be deleted? If yes, then they can be resubmitted by deletion and re-creation anyway.

**Reply by idemus** [reply]
I think it should be simple but complete.
So if a request is rejected it can't be resubmitted but will be kept for overview purposes the whole year. A comment field when rejecting/approving would be good. After year change it could be deleted from database from my point of view.
If circumstances change, etc. it should be possible to create a new request for the same day.
I want to avoid spending too much time with offline 'discussions' for holiday discussions instead of making a fast decision with one click and some text within the workflow.

**Reply by kevinpapst** [reply]
I just uploaded a new version. 
It has a "rejected" state which only allows to delete the entry.

**Comment by idemus** [comment]
Hi Kevin,
thank you for adding the rejected state in the workflow.
I've just tested and it works fine :).
